test/tests/hostmask.js

```diff
@@ -60,3 +67,38 @@ describe("Hostmask", function () {
 		expect(Helper.compareHostmask(a, b)).to.be.true;
 	});
 });
+
+describe("compareWithWildcard", function () {
+	const goodPairs = [
+		["asdf", "asdf"],
+		["a?df*", "asdf"],
+		["*asdf*", "asdf"],
+		["*asdf", "asdf"],
+		["asd?", "asdf"],
+		["asd?*", "asdf"],
+		["*", ""],
+		["**", ""],
+	];
+
+	for (const t of goodPairs) {
+		it(`("${t[0]}", "${t[1]}")`, function () {
+			expect(Helper.compareWithWildcard(t[0], t[1])).to.be.true;
+		});
+	}
+
+	const badPairs = [
+		["asdf", "fdsa"],
+		["a?df*", "adfg"],
+		["?", ""],
+		["?asdf", "asdf"],
+		["?*", ""],
+		["*?*", ""],
```

Comment:
Could you add these cases?

```js
["asd", "asdf"],
["sdf", "asdf"],
["sd", "asdf"],
["", "asdf"],
```

They currently fail. You'll want to add `^` and `$` to the regex to make sure it matches the whole string.

Reply:
Right, nice catch! Thanks